internal/db/schema/migrations/postgres/2/02_vault_credential.up.sql

```diff
@@ -0,0 +1,214 @@
+
+  create table credential_vault_library (
+    public_id wt_public_id primary key,
+    store_id wt_public_id not null
+      constraint credential_vault_store_fk
+        references credential_vault_store (public_id)
+        on delete cascade
+        on update cascade,
+    name wt_name,
+    description wt_description,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+    vault_path text not null
+      constraint vault_path_must_not_be_empty
+        check(length(trim(vault_path)) > 0),
+    constraint credential_vault_library_store_id_name_uq
+      unique(store_id, name),
+    constraint credential_library_fk
+      foreign key (store_id, public_id)
+      references credential_library (store_id, public_id)
+      on delete cascade
+      on update cascade,
+    constraint credential_vault_library_store_id_public_id_uq
+      unique(store_id, public_id)
+  );
+  comment on table credential_vault_library is
+    'credential_vault_library is a table where each row is a resource that represents a vault credential library. '
+    'It is a credential_library subtype and a child table of credential_vault_store.';
+
+  create trigger update_version_column after update on credential_vault_library
+    for each row execute procedure update_version_column();
+
+  create trigger update_time_column before update on credential_vault_library
+    for each row execute procedure update_time_column();
+
+  create trigger default_create_time_column before insert on credential_vault_library
+    for each row execute procedure default_create_time();
+
+  create trigger immutable_columns before update on credential_vault_library
+    for each row execute procedure immutable_columns('public_id', 'store_id','create_time');
+
+  create trigger insert_credential_library_subtype before insert on credential_vault_library
+    for each row execute procedure insert_credential_library_subtype();
+
+  create trigger delete_credential_library_subtype after delete on credential_vault_library
+    for each row execute procedure delete_credential_library_subtype();
+
+  create table credential_vault_token (
+    vault_token text primary key
+      constraint vault_token_must_not_be_empty
+        check(length(trim(vault_token)) > 0),
+    store_id wt_public_id not null
+      constraint credential_vault_store_fk
+        references credential_vault_store (public_id)
+        on delete cascade
+        on update cascade,
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+    accessor text not null
+      constraint credential_vault_token_accessor_uq
+        unique
+      constraint accessor_must_not_be_empty
+        check(length(trim(accessor)) > 0),
+    lease_duration bigint not null
+      constraint lease_duration_must_not_be_negative
+        check(lease_duration >= 0),
+    last_renewal_time wt_timestamp not null
```

Comment:
other `wt_timestamps` you don't have not null?  I actually thought the domain was created with `not null` but see it is not